rpcs3/Emu/Io/usio.cpp

```diff
@@ -203,12 +236,57 @@ void usb_device_usio::translate_input()
 						// Small hit side right
 						std::memcpy(input_buf.data() + 38 + offset, &c_small_hit, sizeof(u16));
 						break;
+					}
+				}
+				else if (button.m_offset == CELL_PAD_BTN_OFFSET_DIGITAL1 && player == 0)
+				{
+					switch (button.m_outKeyCode)
+					{
+					case CELL_PAD_CTRL_SELECT:
+						// Test switch
+						if (!is_test_pressed)
+						{
+							is_test         = !is_test;
+							is_test_pressed = true;
+						}
+						break;
+					case CELL_PAD_CTRL_START:
+						buttons_p1 += TAIKO_BTN_ENTER;
+						break;
+					case CELL_PAD_CTRL_UP:
+						buttons_p1 += TAIKO_BTN_UP;
+						break;
+					case CELL_PAD_CTRL_DOWN:
+						buttons_p1 += TAIKO_BTN_DOWN;
+						break;
+					case CELL_PAD_CTRL_LEFT:
+						buttons_p1 += TAIKO_BTN_SERVICE;
+						break;
+					case CELL_PAD_CTRL_RIGHT:
+						if (!is_coin_pressed)
+						{
+							coin_counter += 1;
+							is_coin_pressed = true;
+						}
+						break;
 					default:
 						break;
 					}
 				}
 			}
 		}
+
+		if (player == 0)
+		{
+			// Like some of namcos other arcade games, taikos test switch is a physical switch and not a button
+			if (is_test)
+			{
+				std::memcpy(input_buf.data() + 0, &button_test, sizeof(u8));
+			}
+
+			std::memcpy(input_buf.data() + 1, &buttons_p1, sizeof(u8));
```

Comment:
Move this out of translate_from_pad

Reply:
otherwise, when the game window gets unfocused, the game will exit the Test Menu automatically.

Reply:
If you actually test it with the game, you will understand what I'm referring to.